calculus.tex

```diff
@@ -0,0 +1,81 @@
+\subsection{Derivatives}
+
+\newcommand{\ddx}{\frac{\partial}{\partial x}}
+\newcommand{\ddX}{\frac{\partial}{\partial X}}
+\newcommand{\ddy}{\frac{\partial}{\partial y}}
+
+Using matrix notation, taking derivatives with respect to vectors or matrices is tricky. With tensors, it's relatively easy \citep{laue+:2018}, and likewise with named tensors.
+
+\paragraph{Definition}
+
+Let $f \colon \reals^\mathcal{S} \rightarrow \reals^\mathcal{T}$. The derivative of $f$ adds $|\mathcal{S}|$ new axes to the return type of $f$, and they must not already be names in $\mathcal{T}$. So if $\mathcal{S} = \name{ax_1} \times \cdots \times \name{ax\sub{r}}$, then for each axis name $\name{ax\sub{i}}$, let $\name{ax\sub{i}'}$ be a new axis name, not in $\mathcal{T}$, and let $\mathcal{S'} = \name{ax_1'} \times \cdots \times \name{ax\sub{r}'}$. If $s = \{\nidx{ax_1}{i_1}, \ldots, \nidx{ax\sub{r}}{i_r}\}$, let $s' = \{\nidx{ax'_1}{i_1}, \ldots, \nidx{ax'\sub{r}}{i_r}\}$.
+
+Then the derivative of $f$ at $x$ is the tensor with shape $\mathcal{S}' \times \mathcal{T}$ such that for all $s \in \rec\mathcal{S}$ and $t \in \rec\mathcal{T}$,
+\[\left[\ddx f(x) \right]_{s',t} = \frac{\partial}{\partial x_{s'}} [f(x)]_t.\]
+
+The primed axis names are inconvenient, but fortunately we usually don't have to work with them explicitly.
```

Comment:
I'm not convinced this is that bad. What if you make this a positive? Something like:

"The primed names give a clear way to distinguish between "input / forward" axes and "output / backward" axes (probably a better name here). In matrix calculus this is typically done with a single "m x n" matrix. This notation allows us to have maintain multiple semantic axes, while representing a tensor derivative."

Reply:
I think 29f5d179e9807aaff0407f11c19c7d47c6f60738 kind of does this